Docstring for Pair

freeboson opened the pull request:
First PR here — followed `CONTRIBUTING.md` for adding a docstring. Not sure if `Pair` needs a blob in the manual,  unless I make a section like "Utility Collections".

Prompted by #23979.

Edit: fix #23979

freeboson committed:
added docstring for Pair and a shoutout in collections.md

freeboson commented:
Note that I only added a name-drop for `Pair` in the manual under the list of collections that implement the iterator protocol.

fredrikekre commented on base/pair.jl:
```diff
@@ -1,5 +1,32 @@
+are stored in the fields `first` and `second`. They can also be accessed via
+iteration.
+
+See also [`Dict`].
```
Since `Dict` has a docstring you can link to it with `@ref` as follows: ``[`Dict`](@ref)``

fredrikekre commented on doc/src/stdlib/collections.md:
```diff
@@ -47,6 +47,7 @@ Fully implemented by:
   * `EachLine`
   * `AbstractString`
   * [`Set`](@ref)
+  * `Pair`
```
Here you can also reference with ``[`Pair`](@ref)``

fredrikekre reviewed (commented):
Great first PR! :+1: 

The docstring also need to a `@docs` block in `doc/src/stdlib` (point 4 here: https://github.com/JuliaLang/julia/blob/master/CONTRIBUTING.md#adding-a-new-docstring-to-base). The most fitting place would probably be in the `doc/src/stdlib/collections.md` file in a location you see fit.

fredrikekre commented on base/pair.jl:
```diff
@@ -1,5 +1,32 @@
+iteration.
+
+See also [`Dict`].
+
```
Should also add `# Examples` on this line, just above the ```` ```jldoctest ```` line.

freeboson committed:
Added Pair to manual in Collections, links in docstr

freeboson commented:
@fredrikekre sorry for the delay. I've added a `@docs` for `Pair` in Collections under a new section "Utility Collections" because that's how the equivalent is categorized in C++'s STL. I've also added the `@ref`s that you suggested.

freeboson committed:
remove trailing newline

fredrikekre commented on doc/src/stdlib/collections.md:
```diff
@@ -265,3 +266,9 @@ Fully implemented by:
   * `Vector` (a.k.a. 1-dimensional [`Array`](@ref))
   * `BitVector` (a.k.a. 1-dimensional [`BitArray`](@ref))
+
+## Utility Collections
```
I think we should move this section to the top of this file, and include the other collections in there too. Specifically `Set`, `IntSet` and `Dict` which are all included in weird places in this file.

freeboson replied:
I'm not sure: Associative and Set-like more or less refers to the protocol of implementing those methods. There isn't really anything like that for Pair.

fredrikekre committed:
whitespace
